azure/defaults.go

```diff
@@ -84,6 +84,14 @@ const (
 	ProviderIDPrefix = "azure://"
 )
 
+const (
+	// CustomHeaderPrefix is the prefix of annotations that enable additional cluster / node pool features.
+	// Whatever follows the prefix will be passed as a header to cluster/node pool creation/update requests.
+	// E.g. add `"custom-headers.azure.dev/UseGPUDedicatedVHD": "true"` annotation to AzureManagedMachinePool CR
+	// to enable creating GPU nodes by the node pool.
+	CustomHeaderPrefix = "cluster-azure.x-k8s.io/custom-header-"
```

Comment:
I don't think we can include a trailing hyphen here, and in fact, I think we can't include a `/` in the prefix as well. According to this document, the `/` is meant to delimit the prefix from the annotation "key" (in a key/value context):

https://kubernetes.io/docs/concepts/overview/working-with-objects/annotations/#syntax-and-character-set

Why not use `azuremanagedcluster.cluster.x-k8s.io` as the prefix? This would yield something like this:

```
kind: AzureManagedMachinePool
metadata:
  annotations:
    azuremanagedcluster.cluster.x-k8s.io/UseGPUDedicatedVHD: true
```

Reply:
`CustomHeaderPrefix` is not meant to be an annotation prefix- it's almost a full annotation and only misses the name of the header (that is a simple string that doesn't contain `/` or `-`). I don't see a conflict between annotation syntax and this implementation. Also this implementation is future proof- it will be possible to enable existing and future preview features this way.

You suggest passing everything that is prefixed with `azuremanagedcluster.cluster.x-k8s.io/` as a API request header?

Reply:
I understand now, thanks for the clarification!